syntax = "proto2";

message Attribute {

    enum Type { STRING = 1; DOUBLE = 2; INT = 3; }

    required Type type = 1;

    required string key = 2;

    optional string str_val = 3;
    optional double double_val = 4;
    optional int32 int_val = 5;

}

message AttributeList {
    repeated Attribute entries = 1;
}
